Documentation/coding-guidelines/breaking-changes.md

```diff
@@ -16,47 +22,86 @@ To help triage breaking changes, we classify them in to four buckets:
 * change in timing/order of events (even when not specified in docs)
-* change in parsing of input and throwing new errors (even if parsing behavior is not specified in the docs)
+* change in parsing of input and throwing new errors (even if parsing behavior
+  is not specified in the docs)
 
 These require judgment: how predictable, obvious, consistent was the behavior?
 
-### Bucket 3: Unlikely Grey Area
-*Change of behavior that customers could have depended on, but probably wouldn't.*
+## Bucket 3: Unlikely Grey Area
+*Change of behavior that customers could have depended on, but probably*
+*wouldn't.*
+
+Examples:
 
-**Examples:**
 * correcting behavior in a subtle corner case
 
-As with type 2 changes, these require judgment: what is reasonable and what’s not?
+As with type 2 changes, these require judgment: what is reasonable and what’s
+not?
+
+## Bucket 4: Clearly Non-Public
+*Changes to surface area or behavior that is clearly internal or non-breaking*
+*in theory, but breaks an app.*
 
-### Bucket 4: Clearly Non-Public
-*Changes to surface area or behavior that is clearly internal or non-breaking in theory, but breaks an app.*
+Examples:
 
-**Examples:**
 * Changes to internal API that break private reflection
 
-It is impossible to evolve a code base without making such changes, so we don't require up-front approval for these, but we will sometimes have to go back and revisit such change if there's too much pain inflicted on the ecosystem through a popular app or library.
+It is impossible to evolve a code base without making such changes, so we don't
+require up-front approval for these, but we will sometimes have to go back and
+revisit such change if there's too much pain inflicted on the ecosystem through
+a popular app or library.
+
+This bucket is painful for the machine-wide .NET Framework, but we do have much
+more latitude here in .NET Core.
+
+## What This Means for Contributors
+
+* All bucket 1, 2, and 3 breaking changes require talking to the repo owners
+  first:
+    - We generally **don't accept** change proposals that are in bucket #1.
+    - We **might accept** change proposals that are in #2 and #3 after a
+      risk-benefit analysis. See below for more details.
+    - We **usually accept** changes that are in bucket #4
+* If you're not sure in which bucket applies to a given change, contact us as
+  well.
+
+### Risk-Benefit Analysis
+
+For buckets #2 and #3 we apply a risk-benefit analysis. It doesn't matter if the
+old behavior is "wrong", we still need to think through the implications. This
+can result in one of the following outcomes: 
+
+* **Accepted with quirking**. Depending on the estimated customer impact, we may
+  decide to "quirk" the behavior. That means, that the new behavior is only
+  exposed to apps that are compiled against the version of .NET Core that
+  introduced the change. In other words, apps running against a shared framework
```

Comment:
That's not my understanding of how `AppContext` works in Core. It's different to Quirks on desktop. In Core, we don't know what version they compiled against. `AppContext` merely provides a way to disable behavior if they need to.